parsl/providers/lsf/lsf.py

```diff
@@ -0,0 +1,223 @@
+    launcher : Launcher
+        Launcher for this provider. Possible launchers include
+        :class:`~parsl.launchers.SingleNodeLauncher` (the default),
+        :class:`~parsl.launchers.SrunLauncher`, or
+        :class:`~parsl.launchers.AprunLauncher`
+     move_files : Optional[Bool]: should files be moved? by default, Parsl will try to move files.
+    """
+
+    def __init__(self,
+                 channel=LocalChannel(),
+                 nodes_per_block=1,
+                 init_blocks=1,
+                 min_blocks=0,
+                 max_blocks=10,
+                 parallelism=1,
+                 walltime="00:10:00",
+                 scheduler_options='',
+                 worker_init='',
+                 project=None,
+                 cmd_timeout=120,
+                 move_files=True,
+                 launcher=SingleNodeLauncher()):
+        label = 'LSF'
+        super().__init__(label,
+                         channel,
+                         nodes_per_block,
+                         init_blocks,
+                         min_blocks,
+                         max_blocks,
+                         parallelism,
+                         walltime,
+                         cmd_timeout=cmd_timeout,
+                         launcher=launcher)
+
+        self.project = project
+        self.move_files = move_files
+        self.scheduler_options = scheduler_options
+        self.worker_init = worker_init
+
+    def _status(self):
+        ''' Internal: Do not call. Returns the status list for a list of job_ids
+
+        Args:
+              self
+
+        Returns:
+              [status...] : Status list of all jobs
+        '''
+        job_id_list = ','.join(self.resources.keys())
+        cmd = "bjobs {0}".format(job_id_list)
+
+        retcode, stdout, stderr = super().execute_wait(cmd)
+        # Execute_wait failed. Do no update
+        if retcode != 0:
+            logger.debug("Updating job status from {} failed with return code {}".format(self.label,
+                                                                                         retcode))
+            return
+
+        jobs_missing = list(self.resources.keys())
+        for line in stdout.split('\n'):
+            parts = line.split()
+            if parts and parts[0] != 'JOBID':
+                job_id = parts[0]
+                status = translate_table.get(parts[2], 'UNKNOWN')
+                self.resources[job_id]['status'] = status
+                jobs_missing.remove(job_id)
+
+        # squeue does not report on jobs that are not running. So we are filling in the
+        # blanks for missing jobs, we might lose some information about why the jobs failed.
+        for missing_job in jobs_missing:
+            if self.resources[missing_job]['status'] in ['PENDING', 'RUNNING']:
+                self.resources[missing_job]['status'] = 'COMPLETED'
+
+    def submit(self, command, blocksize, tasks_per_node, job_name="parsl.auto"):
```

Comment:
Remove blocksize from this (PR #1178 removed it from the submit API) - otherwise blocksize/task_per-node/job_name will get jumbled on invocations

Reply:
Done